upsolver-api/routes/users.js

```diff
@@ -50,4 +58,36 @@ router.post("/register", async (req, res) => {
   }
 });
 
+router.post("/login", async (req, res) => {
+  try {
+    const { username, password } = req.body;
+
+    if (!(username && password)) {
+      res.status(400).json({ error: "[!] Please enter all required fields" });
+    }
+
+    const user = await User.findOne({
+      where: { username: username.toLowerCase() },
+    });
+
+    if (user && (await bcrypt.compare(password, user.password))) {
+      const token = jwt.sign(
+        { user_id: user.id, username: username.toLowerCase() },
+        process.env.TOKEN_KEY,
+        {
+          expiresIn: "2h",
+        }
+      );
+
+      user.update({ token });
+
+      res.status(200).json(user);
+    } else {
+      res.status(400).json({ error: "[!] Invalid Credentials" });
+    }
+  } catch (err) {
+    console.log(err);
```

Comment:
same comment as `/signup` route - do we want to return some useful error to the user as well?

Reply:
Fixed in: https://github.com/cdamezcua/Upsolver/pull/21